src/mqtt/mqtt_test.c

```diff
@@ -1003,7 +1003,7 @@ TEST( MqttTest, MQTT_Connect_LWT )
  */
 TEST( MqttTest, MQTT_ProcessLoop_KeepAlive )
 {
-    uint32_t connectPacketTime = context.lastPacketTime;
```

Comment:
https://github.com/FreeRTOS/coreMQTT/commit/c92e5ff606a16d9dc68b6dd6defcf38129f7925b

Checks that both lastPacketTxTime and lastPacketRxTime do not exceed PACKET_TX_TIMEOUT_MS, PACKET_RX_TIMEOUT_MS, or keepAliveIntervalSec.

Pretty sure this is well covered by unit tests though.

Reply:
They're two different view of test.
In unit-test, it's used to check the behavior inside coreMQTT only.
But in integration test, it also covered end-to-end (device <-> MQTT broker).